lightwood/model/lightgbm.py

```diff
@@ -204,5 +205,12 @@ def __call__(self, ds: EncodedDs) -> pd.DataFrame:
         else:
             decoded_predictions = raw_predictions
 
-        ydf = pd.DataFrame({'prediction': decoded_predictions})
+        if predict_proba and self.ordinal_encoder is not None:
+            predictions = np.hstack([raw_predictions, decoded_predictions.reshape(-1, 1)])
+            cat_labels = [f'__mdb_proba_{label}' for label in self.ordinal_encoder.categories_[0].tolist()]
+            ydf = pd.DataFrame(predictions, columns=cat_labels + ['prediction'])
```

Comment:
I *think* a better solution here is:
```
ydf = pd.DataFrame({'prediction': decoded_predictions}) --- keep this bit instead of deleting
if predict_proba and self.ordinal_encoder is not None:
    for idx, label in for label in enumerate(self.ordinal_encoder.categories_[0].tolist()):
        ydf.insert(len(ydf), label, decoded_predictions[idx]
```

Note: Haven't ran the code, roughly this you might have to change it a bit.

It will be much faster, it's shorter and avoid an `else` *and* it's much easier to understand what it does.